src/realm/object-store/sync/sync_session.cpp

```diff
@@ -651,7 +659,6 @@ void SyncSession::handle_error(sync::SessionErrorInfo error)
             case sync::ProtocolErrorInfo::Action::ApplicationBug:
                 [[fallthrough]];
             case sync::ProtocolErrorInfo::Action::ProtocolViolation:
-                next_state = NextStateAfterError::inactive;
```

Comment:
What was the point of removing this line - do we want to rely on the default state of error/none (based on the isFatal value) for these cases?

Reply:
Exactly. I think it's an oversight from when I refactored error handling in SyncSession. `AplicationBug` and `ProtocolError` should be fatal (cc @kmorkos), so we'll cancel the callbacks. You can argue that before we were pausing the session too, but the user should react on this anyway.